### PR TITLE
Cloudflare Images

### PR DESCRIPTION
**Pull request by matthewpi**
## Description

Adds support for [Cloudflare Images](https://developers.cloudflare.com/images/cloudflare-images)

https://api.cloudflare.com/#cloudflare-images-properties

## Has your change been tested?

- Unit tests
- Against production API, uploading over 40,000 images.

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.

### CODE REVIEW

File: images.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	if err != nil {
+		return Image{}, errors.Wrap(err, errUnmarshalError)
+	}
+	return imageUploadResponse.Result, nil
+}
+
+// Images lists all images.
+//
+// API Reference: https://api.cloudflare.com/#cloudflare-images-list-images
+func (api *API) Images(ctx context.Context, accountID string, pageOpts PaginationOptions) ([]Image, error) {
```
Comment (jacobbednarz) [inline images.go:146]: we should prefix this with `List` as we are fetching all resources.

```suggestion
func (api *API) ListImages(ctx context.Context, accountID string, pageOpts PaginationOptions) ([]Image, error) {
```

**Comment by jacobbednarz** [comment]
> Endpoints are publicly documented, but I am unsure if they are considered stable.

providing they are documented publicly, we can consider the latter part true :)

**Comment by matthewpi** [comment]
> > Endpoints are publicly documented, but I am unsure if they are considered stable.
> 
> providing they are documented publicly, we can consider the latter part true :)

I mentioned that because I was unsure.  I wrote this code back when Images were still in beta and the API documentation at the time had some incorrect responses, but everything seems to be good now.

After quickly looking at the documentation I found I missed some endpoints, so I'm going to see about quickly adding those in as well.

File: images_test.go [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+	Variants: []string{
+		"https://imagedelivery.net/MTt4OTd0b0w5aj/ZxR0pLaXRldlBtaFhhO2FiZGVnaA/hero",
+		"https://imagedelivery.net/MTt4OTd0b0w5aj/ZxR0pLaXRldlBtaFhhO2FiZGVnaA/original",
+		"https://imagedelivery.net/MTt4OTd0b0w5aj/ZxR0pLaXRldlBtaFhhO2FiZGVnaA/thumbnail",
+	},
+	Uploaded: timeMustParse(time.RFC3339, "2014-01-02T02:20:00Z"),
+}
+
+func TestUploadImage(t *testing.T) {
+	setup(UsingAccount("foo"))
```
Comment (jacobbednarz) [inline images_test.go:42]: let's use the `testAccountID` value here instead of "foo".

```suggestion
	setup(UsingAccount(testAccountID))
```

**Comment by jacobbednarz** [comment]
thank you @matthewpi, this is awesome!

**Pull request merged**